python/pyarrow/_dataset.pyx

```diff
@@ -2814,37 +2751,24 @@ cdef class Scanner(_Weakrefable):
         scanner = GetResultValue(builder.get().Finish())
         return Scanner.wrap(scanner)
 
-    def scan(self):
-        """Returns a stream of ScanTasks
-
-        The caller is responsible to dispatch/schedule said tasks. Tasks should
-        be safe to run in a concurrent fashion and outlive the iterator.
+    @property
+    def dataset_schema(self):
+        """The physical schema of the data."""
```

Comment:
Is this necessarily the physical schema? (I mean, it doesn't necessarily correspond to Fragment.physical_schema, since that can be different for different fragments of a dataset)

Reply:
Good point, I aligned it with the C++ docstring.